ivadomed/main.py

```diff
@@ -1,649 +1,214 @@
 
-    if film_bool:  # normalize metadata before sending to network
-        metadata_clustering_models = joblib.load(
-            "./" + context["log_directory"] + "/clustering_models.joblib")
 
-        ds_test = imed_film.normalize_metadata(ds_test,
-                                               metadata_clustering_models,
-                                               context["debugging"],
-                                               context["metadata"],
-                                               False)
+def run_main():
+    if len(sys.argv) <= 1:
+        print("\nivadomed [config.json]\n")
+        return
 
-        one_hot_encoder = joblib.load("./" + context["log_directory"] + "/one_hot_encoder.joblib")
+    with open(sys.argv[1], "r") as fhandle:
+        context = json.load(fhandle)
 
-    if not context["unet_3D"]:
-        print(f"\nLoaded {len(ds_test)} {context['slice_axis']} slices for the test set.")
+    command = context["command"]
+    log_directory = context["log_directory"]
+    if not os.path.isdir(log_directory):
+        print('Creating log directory: {}'.format(log_directory))
+        os.makedirs(log_directory)
     else:
-        print(f"\nLoaded {len(ds_test)} volumes of size {context['length_3D']} for the test set.")
-
-    test_loader = DataLoader(ds_test, batch_size=context["batch_size"],
-                             shuffle=False, pin_memory=True,
-                             collate_fn=imed_loader_utils.imed_collate,
-                             num_workers=0)
-
-    model = torch.load("./" + context["log_directory"] + "/best_model.pt", map_location=device)
-
-    if cuda_available:
-        model.cuda()
-    model.eval()
-
-    # create output folder for 3D prediction masks
-    path_3Dpred = os.path.join(context['log_directory'], 'pred_masks')
-    if not os.path.isdir(path_3Dpred):
-        os.makedirs(path_3Dpred)
-
+        print('Log directory already exists: {}'.format(log_directory))
+
+    # Define device
+    cuda_available, device = define_device(context['gpu'])
+
+    # Get subject lists
+    train_lst, valid_lst, test_lst = get_subdatasets_subjects_list(context["split_dataset"],
+                                                                   context['bids_path'],
+                                                                   log_directory)
+
+    # Get transforms for each subdataset
+    transform_train_params, transform_valid_params, transform_test_params = \
+        get_subdatasets_transforms(context["transformation"])
+    if command == "train":
+        display_selected_transfoms(transform_train_params, dataset_type="training")
+        display_selected_transfoms(transform_valid_params, dataset_type="validation")
+    elif command == "test":
+        display_selected_transfoms(transform_test_params, dataset_type="testing")
+
+    # Loader params
+    loader_params = context["loader_parameters"]
+    extra_params = {"bids_path": context['bids_path'],
```

Comment:
I'm wondering if we should put target_suffix and bids_path directly in loader_parameters in the config_files, since they are loader parameters and this would avoid adding them in the main. There might be a reason they were not included in the first place, that's why I am asking.

Reply:
it's also needed in `cmd_eval` which doesn't use the loader params so maybe that's why?

Reply:
`cmd_eval` was the reason, but we can still extract them from `loader_params` --> happy to change this.

Reply:
done